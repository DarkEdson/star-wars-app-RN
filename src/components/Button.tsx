import React, {ReactElement} from 'react';
import {StyleSheet, TouchableOpacity, Dimensions} from 'react-native';
import {colors} from '../theme';
import Text from './Text';
import {useSelector} from 'react-redux';
import {selectTheme} from '../redux/slices/app.slice';

const {width} = Dimensions.get('screen');

export default function ThemeButtom({
  children,
  onPress,
  testID,
}: {
  children: string | ReactElement;
  onPress?: () => void;
  testID?: string;
}) {
  const currentTheme = useSelector(selectTheme);
  const styles = StyleSheet.create({
    container: {
      backgroundColor: colors[currentTheme].button,
      paddingVertical: 10,
      width: width * 0.9,
      justifyContent: 'center',
      alignItems: 'center',
      marginVertical: 15,
    },
  });
  return (
    <TouchableOpacity
      testID={testID}
      style={styles.container}
      onPress={onPress}>
      <Text bold color={colors[currentTheme].buttonText}>
        {children}
      </Text>
    </TouchableOpacity>
  );
}
