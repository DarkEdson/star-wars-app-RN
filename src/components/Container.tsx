import React, {ReactElement} from 'react';
import {View, StyleSheet, Dimensions, ScrollView} from 'react-native';
import {colors} from '../theme';
import {useSelector} from 'react-redux';
import {selectTheme} from '../redux/slices/app.slice';

const {width, height} = Dimensions.get('window');

export default function Container({
  children,
  scroll,
}: {
  children: ReactElement | ReactElement[];
  scroll?: boolean;
}) {
  const currentTheme = useSelector(selectTheme);
  const styles = StyleSheet.create({
    main: {
      width,
      paddingLeft: width * 0.05,
      paddingRight: width * 0.05,
      height,
      backgroundColor: colors[currentTheme].background,
      paddingTop: 40,
    },
    scrollContainer: {
      height,
      paddingBottom: 180,
    },
  });
  if (scroll) {
    return (
      <View style={styles.main}>
        <View style={styles.scrollContainer}>
          <ScrollView bounces={false} overScrollMode="never">
            {children}
          </ScrollView>
        </View>
      </View>
    );
  }
  return <View style={styles.main}>{children}</View>;
}
