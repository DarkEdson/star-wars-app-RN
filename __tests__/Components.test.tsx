import {it, afterEach, expect} from '@jest/globals';
import 'react-native';
import React from 'react';
import {cleanup, render, waitFor, fireEvent} from './utils/test.utils';
import Button from '../src/components/Button';
import Container from '../src/components/Container';

afterEach(() => {
  cleanup();
});

const onPress = jest.fn();

describe('Renders Container correctly', () => {
  it('Renders Container correctly without scroll', async () => {
    let component;
    const props = {
      children: <></>,
      os: 'android',
    };
    await waitFor(() => {
      component = render(<Container {...props} />, {});
    });

    expect(component!.toJSON()).toMatchSnapshot();
  });

  it('Renders Container correctly with scroll and ios', async () => {
    let component;
    const props = {
      children: <></>,
      scroll: true,
      os: 'ios',
    };
    await waitFor(() => {
      component = render(<Container {...props} />, {});
    });

    expect(component!.toJSON()).toMatchSnapshot();
  });
});

describe('Button functionality', () => {
  it('Renders Button correctly', async () => {
    let component;
    const props = {
      onPress: () => {},
      children: <></>,
    };
    await waitFor(() => {
      component = render(<Button {...props} />, {});
    });

    expect(component!.toJSON()).toMatchSnapshot();
  });

  it('Fire onpress event correctly', async () => {
    let component: any;
    const props = {
      testId: 'testButton',
      onPress: onPress,
      children: 'testButton',
    };
    await waitFor(() => {
      component = render(<Button {...props} />, {});
    });
    fireEvent.press(component.getByText('testButton'));
    expect(onPress).toBeCalled();
  });
});
