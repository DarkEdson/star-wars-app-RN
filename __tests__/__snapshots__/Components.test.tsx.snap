// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button functionality Renders Button correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#9D59FB",
      "justifyContent": "center",
      "marginVertical": 15,
      "opacity": 1,
      "paddingVertical": 10,
      "width": 675,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFF",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 18,
      }
    }
  />
</View>
`;

exports[`Renders Container correctly Renders Container correctly with scroll and ios 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "height": 1334,
      "paddingLeft": 37.5,
      "paddingRight": 37.5,
      "paddingTop": 80,
      "width": 750,
    }
  }
>
  <View
    style={
      {
        "height": 1334,
        "paddingBottom": 180,
      }
    }
  >
    <RCTScrollView
      bounces={false}
      overScrollMode="never"
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;

exports[`Renders Container correctly Renders Container correctly without scroll 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "height": 1334,
      "paddingLeft": 37.5,
      "paddingRight": 37.5,
      "paddingTop": 40,
      "width": 750,
    }
  }
/>
`;
